laika/reports.py

```diff
@@ -1095,7 +1095,36 @@ def create_drive(profile, grant):
     return GoogleDrive(gauth)
 
 
-class DownloadFromGoogleDrive(FileReport):
+class DriveMixin(object):
+    start_timeout = 2
+    max_timeout = 300
+    retry_status_codes = (429,)  # 429 = Too many requests
+
+    def _drive_call(self, method, *args, **kwargs):
+        from pydrive.files import ApiRequestError
+        from googleapiclient.errors import HttpError
+        timeout, limit = self.start_timeout, self.max_timeout
+        while True:
+            try:
+                result = method(*args, **kwargs)
+                return result
+            except (ApiRequestError, HttpError) as e:
+                # ApiRequestError is an IOError for some reason, that wraps
+                # an apiclient.errors.HttpError
+                if isinstance(e, ApiRequestError):
+                    status = e.args[0].resp.status
+                else:
+                    status = e.resp.status
+                if timeout < limit and (status >= 500 or status in self.retry_status_codes):
+                    log = 'An error occured executing %s: %s. Waiting %d seconds.'
+                    logging.info(log, method, e, timeout)
+                    time.sleep(timeout)
+                    timeout = timeout * 2
+                else:
+                    raise
```

Comment:
If API raises an error with below 500 and not in `retry_status_codes` I think you should inform it by raising the catched exception.
```suggestion
                    raise e
```

Reply:
Yeah, it's the default action of `raise`, to reraise the catched exception (so there is no difference between the two in this case).